IE9+ support

tlenex opened the pull request:
- Resolves #45 #29
- IE10 full support
- IE9 partial support: UI has a fixed nav menu and panels/frames width/height can't be resized.

allmarkedup commented:
@tlenex many, many thanks for this! I'll try and check it over over the next few days and then get a new version out with these changes. If I run into any issues I'll let you know.

abelmokadem commented:
Don't feel like I can properly review the code, but this makes fractal usable on IE9 and IE10 👍 Love it!

tlenex commented:
During code review, try to ignore `dist/` directory and you will have much less source code to review. I've built mandelbrot in this commit to be able to use this branch in my own projects.

Should I split the build step and provide simple source change for this PR?

allmarkedup commented:
Just to let you know that I haven't forgotten about this - just a bit behind after the Christmas break :-) I'll hopefully get to review it and get it merged in this week.

tlenex commented:
I might add support for IE8 in the future, but it will be painful. I'll make it in different PR, after this one get merged.

allmarkedup commented on assets/scss/components/_ie-fixes.scss:
```diff
@@ -0,0 +1,95 @@
+    .Frame-panel {
+        overflow: auto;
+    }
+    .Frame-panel--sidebar {
```
@tlenex is there a particular reason why you chose to implement this as a toggle-able, absolutely positioned 'menu' panel instead of a fixed-width sidebar for IE9? I have to say my preference would be with a simple fixed-width sidebar (even if it we lose the show/hide ability). Obviously there is less horizontal space for viewing the components in that case but the overlaid menu doesn't feel right to me (and, as I wasn't expecting it, actually thought it was a bug initially). But I would be interested to hear your thoughts.

tlenex replied:
I choose it just to make if as fast as possible. 
Changing it to fixed sidebar is possible, but I'll need more time to do so

dkhuntrods replied:
@allmarkedup @tlenex What's the status on this, are we waiting to hear if the changes had been made to make it a sidebar?

tlenex replied:
@dkhuntrods Yes, exactly. I'll try to fix it this weekend

allmarkedup commented on src/theme.js:
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+require('babel-polyfill');
```
Is `babel-polyfill` needed here? This is only executed server-side so I'm not sure it's needed. Unless I'm missing something?

allmarkedup reviewed (commented):
IE10 support looks great, I'm not sure about the sidebar in IE9 (see note) but these changes are definitely a big help overall. Thanks!

tlenex commented:
I'll try to make some fixes in this weekend.

I have new idea too. 
To fully support fractal on any browser, we can move backwards to obsolete in HTML5 `<frameset>` and `<frame>` elements. To avoid them beeing obsolete in HTML5, the index page can be in XHTML like this: `<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Frameset//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd">` while child html can be in HTML5. 
What do you think about this @allmarkedup ?

allmarkedup commented:
@tlenex sorry I completely missed your last comment for some reason!

I'll have a think about the frameset idea but to be honest I'd rather avoid it if possible. Are the any specific issues that this would solve that are causing headaches in their current implementation?

tlenex committed:
IE9+ support

Resolves #45 #29
IE10 full support
IE9 partial support: UI has a fixed nav menu and panels/frames width/height can't be resized.

tlenex commented:
Not really, just support for even older browsers

tlenex commented:
I removed the `dist` directory from this PR, to avoid conflicts and removed unnecessary `require()` in `theme.js`

If someone wants to test it, the built `dist/` branch is available here: 
https://github.com/tlenex/mandelbrot/tree/feature/built-ie9-ie10-support

tlenex commented:
isses: #45 #56 https://github.com/frctl/fractal/issues/247

andrewfehrsen commented:
Any update on this? Finding issue with IE 11 as well. Main being that the resizing capability is lost on IE 11. Thanks for all the hard work peeps

patriknil90 commented:
Hi! Tried to exchange the @frctl/mandelbrot folder in node_modules with the content of @tlenex [mandelbrot fork](https://github.com/tlenex/mandelbrot/tree/feature/built-ie9-ie10-support). This makes everything work in edge/explorer when running "fractal start --sync", but doesn't change anything for me when running the fractal site from the dist folder that builds with "fractal build".

Did someone else try to integrate @tlenex solution and got it to work within their fractal project? :) Are there any more steps required? I have installed the babel-polyfill as well.

LeBenLeBen commented:
This merge request has been open since 3 years now and although I think supporting IE 11 would be great, both versions 9 and 10 have been discontinued since and are below 0.3% of market share.

It would not make sense to add specificities to the code base for such browsers today.
